modules/xfeatures2d/src/daisy.cpp

```diff
@@ -0,0 +1,1754 @@
+      return;
+
+    set_image( _image );
+
+    // whole image
+    m_roi = Rect( 0, 0, m_image.cols, m_image.rows );
+
+    // get homography
+    Mat H = m_h_matrix;
+
+    // convert to double if case
+    if ( H.depth() != CV_64F )
+        H.convertTo( H, CV_64F );
+
+    set_parameters();
+
+    initialize_single_descriptor_mode();
+
+    // allocate array
+    _descriptors.create( (int) keypoints.size(), m_descriptor_size, CV_32F );
+
+    // prepare descriptors
+    Mat descriptors = _descriptors.getMat();
+    descriptors.setTo( Scalar(0) );
+
+    // iterate over keypoints
+    // and fill computed descriptors
+    if ( H.empty() )
+      for (int k = 0; k < (int) keypoints.size(); k++)
+      {
+          get_descriptor( keypoints[k].pt.y, keypoints[k].pt.x,
+                          m_use_orientation ? (int) keypoints[k].angle : 0,
+                          &descriptors.at<float>( k, 0 ) );
+      }
+    else
+      for (int k = 0; k < (int) keypoints.size(); k++)
+      {
+          get_descriptor( keypoints[k].pt.y, keypoints[k].pt.x,
+                          m_use_orientation ? (int) keypoints[k].angle : 0,
+                          &H.at<double>( 0 ), &descriptors.at<float>( k, 0 ) );
+      }
+
+}
+
+// full scope with roi
+void DAISY_Impl::compute( InputArray _image, Rect roi, OutputArray _descriptors )
+{
+    // do nothing if no image
+    if( _image.getMat().empty() )
+      return;
+
+    CV_Assert( m_h_matrix.empty() );
+    CV_Assert( ! m_use_orientation );
+
+    set_image( _image );
+
+    m_roi = roi;
+
+    set_parameters();
+    initialize_single_descriptor_mode();
+
+    // compute full desc
+    compute_descriptors();
+    normalize_descriptors();
+
+    Mat descriptors = _descriptors.getMat();
+    descriptors = m_dense_descriptors;
+
+    release_auxiliary();
+}
+
+// full scope
+void DAISY_Impl::compute( InputArray _image, OutputArray _descriptors )
+{
+    // do nothing if no image
+    if( _image.getMat().empty() )
+      return;
+
+    CV_Assert( m_h_matrix.empty() );
+    CV_Assert( ! m_use_orientation );
+
+    set_image( _image );
+
+    // whole image
+    m_roi = Rect( 0, 0, m_image.cols, m_image.rows );
+
+    set_parameters();
+    initialize_single_descriptor_mode();
+
+    // compute full desc
+    compute_descriptors();
+    normalize_descriptors();
+
+    Mat descriptors = _descriptors.getMat();
+    descriptors = m_dense_descriptors;
+
+    release_auxiliary();
+}
+
+// constructor
+DAISY_Impl::DAISY_Impl( float _radius, int _q_radius, int _q_theta, int _q_hist,
+             int _norm, InputArray _H, bool _interpolation, bool _use_orientation )
+           : m_rad(_radius), m_rad_q_no(_q_radius), m_th_q_no(_q_theta), m_hist_th_q_no(_q_hist),
+             m_nrm_type(_norm), m_disable_interpolation(_interpolation), m_use_orientation(_use_orientation)
+{
+
+    m_image = 0;
+
+    m_descriptor_size = 0;
+    m_grid_point_number = 0;
+
+    m_grid_points.release();
```

Comment:
the .release() calls are not necessary - all the matrices are empty at the beginning.